File: applications/bot/src/events/GuildCreate.ts
import { Events, type Guild } from "discord.js";

import ConfigManager from "#config/ConfigManager.js";
import EventListener from "#events/EventListener.js";

export default class GuildCreate extends EventListener {
	constructor() {
		super(Events.GuildCreate);
	}

	async execute(guild: Guild): Promise<void> {
		// Trigger the creation/loading of the guild config.
		ConfigManager.getGuildConfig(guild.id);
	}
}

File: backend/src/events/guild_create.rs
use poise::serenity_prelude as serenity;
use tracing::info;
use crate::Data;

/// Handles the GuildCreate event.
///
/// - Triggers guild config creation/loading.
pub async fn handle(
    guild: &serenity::Guild,
    is_new: &Option<bool>,
    data: &Data,
) {
    let _ = data.config_manager.get_guild_config(&data.db, guild.id).await;

    if is_new.unwrap_or(false) {
        info!("Joined new guild: {} ({})", guild.name, guild.id);
    }
}
